src/warning/warning-printer.ts

```diff
@@ -137,7 +137,7 @@ function getSquiggleUnderline(
     const endColumn = sourceRange.end.line === sourceRange.start.line ?
         sourceRange.end.column :
         lineText.length;
-    const prefix = ' '.repeat(startColumn);
+    const prefix = lineText.replace(/[^\t]/g, ' ').substr(0, startColumn);
```

Comment:
This is close, but I don't think this is quite right. prefix must be only whitespace, and `startColumn` might be after non-whitespace characters. E.g. suppose we want to underline MyElem in this line:

```javascript
class MyElem extends HTMLElement {
```

I think the simplest correct solution would be to match tab character for tab character of the original input. Something like:

```javascript
const prefix = Array.from(
    lineText.slice(0, startColumn)).map((c) => c === '\t' ? '\t' : ' ').join('');
```